dist/radar_client.js

```diff
@@ -480,41 +688,49 @@ Client.prototype._restore = function() {
   }
 };
 
-Client.prototype._sendMessage = function(message) {
-  var memorized = this._memorize(message);
-  this.emit('message:out', message);
+Client.prototype._sendMessage = function(request) {
+  var memorized = this._memorize(request),
+      ack = request.getAttr('ack');
+
+  this.emit('message:out', request.getMessage());
 
   if (this._socket && this.manager.is('activated')) {
-    this._socket.sendPacket('message', JSON.stringify(message));
+    this._socket.sendPacket('message', JSON.stringify(request.getMessage()));
   } else if (this._isConfigured) {
     this._restoreRequired = true;
-    if (!memorized || message.ack) {
-      this._queuedMessages.push(message);
+    if (!memorized || ack) {
+      this._queuedMessages.push(request);
     }
     this.manager.connectWhenAble();
   }
 };
 
 Client.prototype._messageReceived = function (msg) {
-  var message = JSON.parse(msg);
+  var message = JSON.parse(msg),
+      op = Response.getAttr(message, 'op'),
+      to = Response.getAttr(message, 'to');
```

Comment:
This is weird. Let's do
`
response = new Response(message),
op = response.getAttr('op'),
`
here. That should also get rid of these class methods.

here.